app/src/main/java/com/example/reduxtestapp/di/Modules.kt

```diff
@@ -75,4 +78,8 @@ val sharedModule = module {
     single<Logger> {
         DefaultLogger
     }
+
+    single<CacheManager<List<CountryDto>>>{
+        CacheManagerImpl()
+    }
```

Comment:
Koin can't differentiate generic types in runtime so CacheManager will have to be de-generified, we can discuss offline